include/sys/zfs_znode.h

```diff
@@ -258,6 +258,13 @@ zfs_inherit_projid(znode_t *dzp)
 #define	ZTOZSB(znode)	((zfsvfs_t *)(ZTOI(znode)->i_sb->s_fs_info))
 #define	ITOZSB(inode)	((zfsvfs_t *)((inode)->i_sb->s_fs_info))
 
+#define	ZTOTYPE(zp)	(ZTOI(zp)->i_mode)
+#define	ZTOGID(zp) (ZTOI(zp)->i_gid)
+#define	ZTOUID(zp) (ZTOI(zp)->i_uid)
+#define	ZTONLNK(type) (ZTOI(zp)->i_nlink)
+#define	Z_ISBLK(type) S_ISBLK(type)
+#define	Z_ISCHR(type) S_ISCHR(type)
+#define	Z_ISLNK(type) S_ISLNK(type)
```

Comment:
Let's use either `mode` or `type` consistently for all `Z_IS*` macros.  Since you've added a `ZTOTYPE` macro, `type` probably makes the most sense.